src/mr/coordinator.go

```diff
@@ -21,14 +21,26 @@ type Coordinator struct {
 //
 // the RPC argument and reply types are defined in rpc.go.
 func (c *Coordinator) RequestTask(args *RequestTaskArgs, reply *RequestTaskReply) error {
-	hasUncompleteMapTasks := false
+	// I think I just had flag in the Coordinator struct for this
+	// I also did the coordinator pretty differently to you
+	// I had a main orchestration thread which 
+	// 1. queued up the map tasks
+	// 2. waited until the map tasks were done
+	// 3. queued up the reduce tasks
```

Comment:
This is how I had it at the start, I don't remember why I switched to creating all of the reduce tasks up front 😅